docs/FHIRSearchConfiguration.md

```diff
@@ -0,0 +1,222 @@
+
+#### 1.1.1 Search Parameters Configuration: extension-search-parameters.json 
+To configure the FHIR server with one or more custom search parameters, one must create a file called `extension-search-parameters.json` and populate the contents with a Bundle of `SearchParameter` resources.
+
+The `fhir-search` module requires the [expression](https://www.hl7.org/fhir/r4/searchparameter-definitions.html#SearchParameter.expression),  [type](https://www.hl7.org/fhir/r4/searchparameter-definitions.html#SearchParameter.type) and [code](https://www.hl7.org/fhir/r4/searchparameter-definitions.html#SearchParameter.code) be set, as in the following example:
+```
+{
+   "resourceType": "Bundle",
+   "id": "searchParams",
+   "meta": {
+      "lastUpdated": "2019-07-12T22:37:54.724+11:00"
+   },
+   "type": "collection",
+   "entry": [{
+      "fullUrl": "http://ibm.com/fhir/SearchParameter/Patient-favorite-color",
+      "resource": {
+         "resourceType": "SearchParameter",
+         "id": "Patient-favorite-color",
+         "url": "http://ibm.com/fhir/SearchParameter/Patient-favorite-color",
+         "version": "4.0.0",
+         "name": "favorite-color",
+         "status": "draft",
+         "experimental": false,
+         "date": "2018-12-27T22:37:54+11:00",
+         "publisher": "IBM Server for HL7 FHIR Test",
+         "contact": [{
+            "telecom": [{
+               "system": "url",
+               "value": "http://ibm.com/fhir"
+            }]
+         },
+         {
+            "telecom": [{
+               "system": "url",
+               "value": "http://ibm.com/fhir"
+            }]
+         }],
+         "description": "the patient's favorite color",
+         "code": "favorite-color",
+         "base": ["Patient"],
+         "type": "string",
+         "xpathUsage": "normal",
+         "xpath": "f:Patient/f:extension[@url='http://ibm.com/fhir/extension/Patient/favorite-color']/f:valueString",
+         "expression": "Patient.extension.where(url='http://ibm.com/fhir/extension/Patient/favorite-color').value",
+         "multipleOr": true,
+         "multipleAnd": true,
+         "modifier": []
+      }
+   }
+}
+```
+
+A few things to note are: 
+- The Search Parameter specification includes an XPath entry.  The XPath entry is included in the example, however added for completeness during extraction. 
+- The Search Parameter uses `value` which applies to Choice data types which are determined based on the Search Parameter type and the Resource's data type. 
```

Comment:
should this mention the valuetypes config file?

Reply:
done

Reply:
Done